frame/contracts/src/exec.rs

```diff
@@ -1230,6 +1234,11 @@ where
 		});
 		Ok(())
 	}
+
+	fn reentrant_count(&mut self) -> u32 {
+		let id: &AccountIdOf<Self::T> = &self.top_frame().account_id;
+		(self.frames().filter_map(|f| Some(&f.account_id == id && f.delegate_caller.is_none())).count() - 1).try_into().unwrap()
```

Comment:
Better first do more common check `f.delegate_caller.is_none()` and after `&f.account_id == id`

Reply:
Also maybe the count of frames should affect the weight of the function.